simpleGraph/toGiraf.go

```diff
@@ -81,9 +83,91 @@ func GraphSmithWatermanToGiraf(gg *SimpleGraph, read *fastq.FastqBig, seedHash m
 	if !currBest.PosStrand {
 		fastq.ReverseQualUint8Record(currBest.Qual)
 	}
+	//end := time.Now()
+	//fmt.Println("Read Time:", end.Nanosecond() - start.Nanosecond())
+	//start = time.Now()
+	currBest.Aln = GirafToExplicitCigar(&currBest, gg)
+	//end = time.Now()
+	//fmt.Println("Cigar Time:", end.Nanosecond() - start.Nanosecond())
+	//fmt.Println()
 	return &currBest
 }
 
+func GirafToExplicitCigar(giraf *giraf.Giraf, graph *SimpleGraph) []*cigar.Cigar {
+	var answer []*cigar.Cigar
+	var seqIdx, refIdx, pathIdx int
+	refIdx = giraf.Path.TStart
+	var k, runLenCount int64
+
+	if giraf.Aln[0].Op == '*' {
+		return nil
```

Comment:
Do we want this to be nil?  '*' means unaligned read, right?  Do we want to keep the cigar as a '*' for unaligned or should we do nil?  This should probably be consistent across all cigars, right?

Reply:
yes, this is not consistent throughout the code base but i think the slice of cigars should be set to nil if it is unaligned. When we print (to string and/or to file) we check if == nil, print *